src/reanimated2/threads.ts

```diff
@@ -27,6 +27,9 @@ export function runOnJS<A extends any[], R>(
   fun: ComplexWorkletFunction<A, R>
 ): (...args: A) => void {
   'worklet';
+  if (fun.__remoteFunction) {
+    fun = fun.__remoteFunction;
+  }
```

Comment:
Can we add some comment what happens here, what's  `fun`, what's `__remoteFunction` etc.?